go/vt/vtadmin/api.go

```diff
@@ -555,3 +602,152 @@ func (api *API) getClustersForRequest(ids []string) ([]*cluster.Cluster, []strin
 
 	return clusters, ids
 }
+
+// VTExplain is part of the vtadminpb.VTAdminServer interface.
+func (api *API) VTExplain(ctx context.Context, req *vtadminpb.VTExplainRequest) (*vtadminpb.VTExplainResponse, error) {
+	span, ctx := trace.NewSpan(ctx, "API.VTExplain")
+	defer span.Finish()
+
+	if req.Cluster == "" {
```

Comment:
This parameter validation feels a bit clumsy so I'm open to suggestions.

Reply:
> It would be really cool if there was a good way to split out a mux route into ${parameter validation} + ${logic} phases

Yeah, something along these lines would be useful. We should, at some point, revisit the HTTP response envelopes (particularly for errors) that we're using internally. I think there are ways we can improve on what we have. Or.. we could try gRPC web again, lol.

> we can also ....................... add a more typed interface to vtexplain :llama_hands:

I really want this. The string marshalling is easy to mess up + is a bit tedious to debug.